app/views/users/_form.html.erb

```diff
@@ -0,0 +1,22 @@
+<% if @user.errors.any? %>
+    <div>The form contains <%= pluralize(@user.errors.count, "error") %>.</div>
+    <ul>
+        <% @user.errors.full_messages.each do |msg| %>
+        <li><%= msg %></li>
+        <% end %>
+    </ul>
+<% end %>
```

Comment:
### Optional
- [ ] There should be NO application logic in the views nor helpers (no Post.new, Comment.create, current_user.friends.find_by(...)...). Everything connected with data processing should be encapsulated in models and used in controllers.

 - [ ] View-related logic can be encapsulated in View helpers (app/helpers/*_helper.rb)

 - [ ] There should be absolutely NO logic in views.

Reply:
Thanks for the tip, logic in views is a bad practice in Rails, could you give me an idea of how to fulfill that Odin requirement by how to **display a list of the error messages that are attached to the failed model object when it fails validations.**

![30ee0294-6084-4f00-a9cc-6da212333385](https://user-images.githubusercontent.com/24830039/119823456-24499880-bef5-11eb-8bc8-ab6ac3e18464.png)